### PR TITLE
Update blog-starter example

### PR DESCRIPTION
**Pull request by timneutkens**
Based on one of the CMS examples, but with Markdown from the filesystem as datasource 💯 

Left to be done:
- Readme guide

## When creating more examples for other Headless CMSs, we need to modify the following:

- [x] `README.md` (the most time consuming part)
- [x] `package.json` - Update `name` and remove `react-datocms`
- [x] `.env.example` - Use different env variable names
- [x] `next.config.js` - Use different env variable names
- [x] `lib/api.js`
- [x] `now.json`
- [x] `pages/index.js` - Update `getStaticProps`
- [x] `pages/posts/[slug].js` - Update `getStaticProps` and `getStaticPaths`
- [x] `pages/api/preview.js` - Update the secret environment variable (`NEXT_EXAMPLE_CMS_DATOCMS_PREVIEW_SECRET`)
- [x] `lib/constants.js` - ogp image is created on https://og-image.now.sh ([config](https://user-images.githubusercontent.com/992008/76175962-1e3e8f00-616c-11ea-954e-f45896caa7e3.png))
- [x] `lib/markdownToHtml.js` - DatoCMS sends down markdown so we had to convert it to HTML, but might not be necessary for other CMSs
- [x] `components/cover-image.js` - Uses `react-datocms` to render progressive images.

Also, add to the "More Examples" section on:

- [x] `docs/basic-features/pages.md`
- [x] `docs/basic-features/data-fetching.md`
- [ ] `docs/advanced-features/preview-mode.md`

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| buildDuration | 9.4s | 9.8s | ⚠️ +424ms |
| nodeModulesSize | 56.6 MB | 56.6 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.77 kB | 5.77 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..8eb0.js gzip | 9.78 kB | 9.78 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.4 kB | 55.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.78 kB | 4.78 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.71 kB | 6.71 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _app.js gzip | 1.09 kB | 1.09 kB | ✓ |
| _error.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.39 kB | 7.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _app.module.js gzip | 594 B | 594 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.28 kB | 5.28 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| index.html gzip | 916 B | 916 B | ✓ |
| link.html gzip | 925 B | 925 B | ✓ |
| withRouter.html gzip | 913 B | 913 B | ✓ |
| Overall change | 2.75 kB | 2.75 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| buildDuration | 9.5s | 9.5s | -88ms |
| nodeModulesSize | 56.6 MB | 56.6 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.77 kB | 5.77 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..8eb0.js gzip | 9.78 kB | 9.78 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.4 kB | 55.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.78 kB | 4.78 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.71 kB | 6.71 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _app.js gzip | 1.09 kB | 1.09 kB | ✓ |
| _error.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.39 kB | 7.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _app.module.js gzip | 594 B | 594 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.28 kB | 5.28 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | timneutkens/next.js add/update-blog-starter | Change |
| - | - | - | - |
| _error.js gzip | 293 kB | 294 kB | ⚠️ +1.16 kB |
| 404.html gzip | 1.32 kB | 1.32 kB | ✓ |
| hooks.html gzip | 957 B | 957 B | ✓ |
| index.js gzip | 294 kB | 294 kB | -10 B |
| link.js gzip | 301 kB | 301 kB | ⚠️ +139 B |
| routerDirect.js gzip | 300 kB | 300 kB | ⚠️ +38 B |
| withRouter.js gzip | 300 kB | 299 kB | -486 B |
| Overall change | 1.49 MB | 1.49 MB | ⚠️ +836 B |

</details>

</details>

**Comment by sandys** [comment]
is it possible to avoid  https://og-image.now.sh ? anything like https://www.gatsbyjs.org/packages/gatsby-remark-images/ that can process offline ?

**Comment by timneutkens** [comment]
> is it possible to avoid og-image.now.sh ? anything like gatsbyjs.org/packages/gatsby-remark-images that can process offline ?

og-image.now.sh is just a og-image generator we use at ZEIT, it has nothing to do with image processing in particular. You already have an open feature request and an open GitHub discussions thread so doesn't seem relevant to discuss that here.

**Review by chibicode — APPROVED** [review]

**Comment by chibicode** [comment]
Looks good, but which account is https://next-blog-starter.now.sh/ deployed to? I tweaked CSS and made it display excerpt, so it needs to be re-deployed.

**Review by lfades — APPROVED** [review]

File: docs/advanced-features/preview-mode.md [hunk 1]
```diff
@@ -153,6 +153,7 @@ https://<your-site>/api/preview?secret=<token>&slug=<path>
 
 Take a look at the following examples to learn more:
 
+- [Blog Starter using markdown files](https://github.com/zeit/next.js/tree/canary/examples/blog-starter) ([Demo](https://next-blog-starter.now.sh/))
```
Comment (timneutkens) [inline docs/advanced-features/preview-mode.md]:
```suggestion
```

**Comment by sandys** [comment]
@timneutkens thank you for this update! would it be possible for you to illustrate pagination in here ?
there was an older question that got closed on pagination - https://github.com/zeit/next.js/issues/8197

it would be really helpful to see how that works with the new apis.

File: docs/advanced-features/preview-mode.md (hunk 1 above)
Comment (chibicode) [reply docs/advanced-features/preview-mode.md]: Oops sorry I didn't realize!

**Comment by brandonpittman** [comment]
@timneutkens Will there be another example that uses MDX instead of plain Markdown?

**Pull request merged**